### PR TITLE
Add BayesianLSTM.ipynb example

### PR DESCRIPTION
**Pull request by pbischoff**
Added example to help beginners adopt `edward2` for BNN with keras. Based on Issue #474.

### CODE REVIEW

**Review by dustinvtran — CHANGES_REQUESTED** [review]
Thanks Peter! This is a super useful contribution. I added some minor comments.

**Comment by pbischoff** [comment]
Thank you for reviewing, @dustinvtran! I committed the requested changes.

File: examples/notebooks/BayesianLSTM.ipynb [hunk 1]
```diff
@@ -0,0 +1,626 @@
...
+   "metadata": {},
+   "source": [
+    "#### Analysing the impact of hyperparameters"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The followin plot shows the test accuracy over the hyperparameters. It's clear that scaling the loss of the bayesian layers in the model is necessary to reach convergence.\n",
```
[the remark this answers is not in the document]
Comment (pbischoff) [reply examples/notebooks/BayesianLSTM.ipynb]: fixed typo

**Review by dustinvtran — APPROVED** [review]

**Comment by dustinvtran** [comment]
Approved PR! I'm waiting on @dusenberrymw to confirm the changes before merging.

File: examples/notebooks/BayesianLSTM.ipynb [hunk 2]
```diff
@@ -0,0 +1,625 @@
...
+    "epochs = 40"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Training a baseline LSTM\n",
+    "This is our baseline model. It's a simple RNN with a single layer of LSTM cells.  \n",
+    "As it would be expected the model cann quickly be fitted to differentiate the valid from the invalid strings."
```
[the remark this answers is not in the document]
Comment (pbischoff) [reply examples/notebooks/BayesianLSTM.ipynb:135]: Sorry for those typos... Going to check for them myself again before pushing the next commit :)

File: examples/notebooks/BayesianLSTM.ipynb [hunk 3]
```diff
@@ -0,0 +1,625 @@
...
+   "metadata": {},
+   "source": [
+    "#### Analysing the impact of hyperparameters"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following plot shows the test accuracy over the hyperparameters. It's clear that scaling the loss of the bayesian layers in the model is necessary to reach convergence.\n",
```
[the remark this answers is not in the document]
Comment (pbischoff) [reply examples/notebooks/BayesianLSTM.ipynb:439]: I would add the following text. I did make it "less mathematical" though (I'm  definitely not an expert on the mathematics of BNN myself and I think it's easier to understand for someone who possibly hasn't heard of KL divergence before). Happy to include more details obviously if they are needed for the explanation to be correct! Please let me know, what you think! 

*What is the correct scaling factor for the regularizers?*

BNN use the Kullback-Leibler Divergence (KL-divergence) as a regularization term to minimize the difference between the prior distribution $p$ and the approximated posterior distribution $q$ over the parameters of the network. This divergence term is written as $KL(q||p)$.  

The KL-divergence should be included once for an entire dataset. Due to the implementation  in `tensorflow` the regularization term is added to the value of the loss function after every batch though, leading to a larger KL-divergence term compared to the loss. To bring both terms on the same scale, we apply a scaling factor < 1 to the KL-divergence. The loss for each batch is divided by the batch size and therefore $KL(q||p)$ (which is in relation to the complete dataset) is divided by the number of training samples (in this case `x_train.shape[0]`).

[the remark this answers is not in the document]
Comment (dusenberrymw) [reply examples/notebooks/BayesianLSTM.ipynb:439]: That sounds great, thanks!

**Comment by Fatigerrr** [comment]
Hello，I have a question about BayesianLSTM。In my code，I use ed.layers.LSTMCellFlipout，but I found that it seemed to make the neural network unable to learn anything, and I strictly followed the code in the example. Could you give me some advice
